### ISSUE TITLE
Compilation error

### ISSUE THREAD

**Issue by rzeldent** [luisllamasbinaburo/Arduino-ReactiveArduino#7]
Hi Luis,

This looks amazing and wanted to try it on an ESP32 using PlatformIo!
However, I do get compilation errors. :-(

I looks like there are some templates overlapping? There are some problems in the files ObservableTimerMillis, ObservableTimerMicros and ObservableIntervalMicros. Could you take a look why? I attached a zip with the platformIo Project.

Looking forward to use your library and kind regards,

Rene

Processing esp32dev (platform: espressif32; board: esp32dev; framework: arduino)
-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
Removing unused dependencies...
Verbose mode can be enabled via `-v, --verbose` option
CONFIGURATION: https://docs.platformio.org/page/boards/espressif32/esp32dev.html
PLATFORM: Espressif 32 (5.1.0) > Espressif ESP32 Dev Module
HARDWARE: ESP32 240MHz, 320KB RAM, 4MB Flash
DEBUG: Current (cmsis-dap) External (cmsis-dap, esp-bridge, esp-prog, iot-bus-jtag, jlink, minimodule, olimex-arm-usb-ocd, olimex-arm-usb-ocd-h, olimex-arm-usb-tiny-h, olimex-jtag-tiny, tumpa)
PACKAGES:
 - framework-arduinoespressif32 @ 3.20004.220818 (2.0.4)
 - tool-esptoolpy @ 1.30300.0 (3.3.0)
 - toolchain-xtensa-esp32 @ 8.4.0+2021r2-patch3
LDF: Library Dependency Finder -> https://bit.ly/configure-pio-ldf
LDF Modes: Finder ~ chain, Compatibility ~ soft
Found 33 compatible libraries
Scanning dependencies...
Dependency Graph
|-- ReactiveArduino @ 1.0.0
Building in release mode
Compiling .pio\build\esp32dev\src\main.cpp.o
Generating partitions .pio\build\esp32dev\partitions.bin
Compiling .pio\build\esp32dev\FrameworkArduino\Esp.cpp.o
Compiling .pio\build\esp32dev\FrameworkArduino\FirmwareMSC.cpp.o
Compiling .pio\build\esp32dev\FrameworkArduino\FunctionalInterrupt.cpp.o
Compiling .pio\build\esp32dev\FrameworkArduino\HWCDC.cpp.o
Compiling .pio\build\esp32dev\FrameworkArduino\HardwareSerial.cpp.o
Compiling .pio\build\esp32dev\FrameworkArduino\IPAddress.cpp.o
Compiling .pio\build\esp32dev\FrameworkArduino\IPv6Address.cpp.o
Compiling .pio\build\esp32dev\FrameworkArduino\MD5Builder.cpp.o
Compiling .pio\build\esp32dev\FrameworkArduino\Print.cpp.o
Compiling .pio\build\esp32dev\FrameworkArduino\Stream.cpp.o
Compiling .pio\build\esp32dev\FrameworkArduino\StreamString.cpp.o
Compiling .pio\build\esp32dev\FrameworkArduino\Tone.cpp.o
Compiling .pio\build\esp32dev\FrameworkArduino\USB.cpp.o
Compiling .pio\build\esp32dev\FrameworkArduino\USBCDC.cpp.o
Compiling .pio\build\esp32dev\FrameworkArduino\USBMSC.cpp.o
Compiling .pio\build\esp32dev\FrameworkArduino\WMath.cpp.o
Compiling .pio\build\esp32dev\FrameworkArduino\WString.cpp.o
In file included from .pio/libdeps/esp32dev/ReactiveArduino/src/Observables/Observables.h:26,
                 from .pio/libdeps/esp32dev/ReactiveArduino/src/ReactiveArduinoLib.h:21,
                 from src/main.cpp:3:
**.pio/libdeps/esp32dev/ReactiveArduino/src/Observables/ObservableTimerMillis.h:45:1: error: redeclaration of 'ObservableTimerMillis<T>::ObservableTimerMillis(long unsigned int, long unsigned int)' may not have default arguments [-fpermissive]**
 ObservableTimerMillis<T>::ObservableTimerMillis(unsigned long interval, unsigned long delay = 0)
 ^~~~~~~~~~~~~~~~~~~~~~~~
In file included from .pio/libdeps/esp32dev/ReactiveArduino/src/Observables/Observables.h:27,
                 from .pio/libdeps/esp32dev/ReactiveArduino/src/ReactiveArduinoLib.h:21,
                 from src/main.cpp:3:
**.pio/libdeps/esp32dev/ReactiveArduino/src/Observables/ObservableTimerMicros.h:45:1: error: redeclaration of 'ObservableTimerMicros<T>::ObservableTimerMicros(long unsigned int, long unsigned int)' may not have default arguments [-fpermissive]**
 ObservableTimerMicros<T>::ObservableTimerMicros(unsigned long interval, unsigned long delay = 0)
 ^~~~~~~~~~~~~~~~~~~~~~~~
In file included from .pio/libdeps/esp32dev/ReactiveArduino/src/Observables/Observables.h:29,
                 from .pio/libdeps/esp32dev/ReactiveArduino/src/ReactiveArduinoLib.h:21,
                 from src/main.cpp:3:
**.pio/libdeps/esp32dev/ReactiveArduino/src/Observables/ObservableIntervalMicros.h:45:1: error: redeclaration of 'ObservableIntervalMicros<T>::ObservableIntervalMicros(long unsigned int, long unsigned int)' may not have default arguments [-fpermissive]**
 ObservableIntervalMicros<T>::ObservableIntervalMicros(unsigned long interval, unsigned long delay = 0)
 ^~~~~~~~~~~~~~~~~~~
[ReactiveTest.zip](https://github.com/luisllamasbinaburo/Arduino-ReactiveArduino/files/10369015/ReactiveTest.zip)
~~~~~~~~
src/main.cpp:7:29: error: expected primary-expression before 'unsigned'
 auto timer = IntervalMillis<unsigned long>(500);
 ```

**Comment by rzeldent** [comment]
Created a pull request. https://github.com/luisllamasbinaburo/Arduino-ReactiveArduino/pull/8
Please take a look and merge it...

Kind regards,

Rene

**Comment by rzeldent** [comment]
Issue is resolved with the update to version 2.0! A big improvement.
Also new library for platformIO is published at the  [PlatformIO registry](https://registry.platformio.org/libraries/luisllamasbinaburo/ReactiveArduino)!

Thanks Luis!
